### ISSUE TITLE
Add stage 2 reductions to the diffy g solver for involutive functions

### ISSUE THREAD

**Issue by msparapa** [Rapid-Design-of-Systems-Laboratory/beluga#127]
After #126, the next step is to implement stage 2 of the reduction process for involutive functions.

**Comment by msparapa** [comment]
Reopening after #201 rewrite.

**Comment by msparapa** [comment]
@SeanMatthewNolan Your suggestion earlier today worked! I perused a bit through [this](http://farside.ph.utexas.edu/teaching/plasma/Plasmahtml/node19.html) and found that by equating the Poincare invariant from before and after the transformation, then root-solving and integrating, I found a convenient expression for the `q`'s. The `dq` in `pdq = PdQ` equaling `dx + dy` wasn't my bias of coordinates after all, its the flatted symmetry as given by either the user or Noether's theorem so it's well defined either way. I ran brachistochrone with a ['1', '1', '0'] symmetry and it actually correctly solved, but it took like 49 seconds. I suspect it's primarily because I don't have a convenient way of compiling a function that takes `x => q` and `p => c` numerically and so the initial guess is way off. I'll need to hold off on that until we get your codegen improvements.
